From: Andre Heider <a.heider@gmail.com>
Date: Tue, 9 Feb 2021 07:37:48 +0100
Subject: meson: fix host cpu setting

CONFIG_TARGET_SUBTARGET is bogus for this, it can contain values such as
'nand' or 'tiny', use the real cpu type instead.

Fixes #14662.

Signed-off-by: Andre Heider <a.heider@gmail.com>

diff --git a/devel/meson/meson.mk b/devel/meson/meson.mk
index b855e267d40d08278f97c415ab96a8b9db0e733c..ac74bd5bf50263a33b36794c27d192886b70001f 100644
--- a/devel/meson/meson.mk
+++ b/devel/meson/meson.mk
@@ -52,6 +52,13 @@ else
 MESON_ARCH:=$(CONFIG_ARCH)
 endif
 
+# this is undefined for just x64_64
+ifeq ($(origin CPU_TYPE),undefined)
+MESON_CPU:="generic"
+else
+MESON_CPU:="$(CPU_TYPE)$(if $(CPU_SUBTYPE),+$(CPU_SUBTYPE))"
+endif
+
 define Meson
 	$(2) $(STAGING_DIR_HOST)/bin/$(PYTHON) $(MESON_DIR)/meson.py $(1)
 endef
@@ -81,7 +88,7 @@ define Meson/CreateCrossFile
 		-e "s|@CXXFLAGS@|$(foreach FLAG,$(TARGET_CXXFLAGS) $(EXTRA_CXXFLAGS) $(TARGET_CPPFLAGS) $(EXTRA_CPPFLAGS),'$(FLAG)',)|" \
 		-e "s|@LDFLAGS@|$(foreach FLAG,$(TARGET_LDFLAGS) $(EXTRA_LDFLAGS),'$(FLAG)',)|" \
 		-e "s|@ARCH@|$(MESON_ARCH)|" \
-		-e "s|@CPU@|$(CONFIG_TARGET_SUBTARGET)|" \
+		-e "s|@CPU@|$(MESON_CPU))|" \
 		-e "s|@ENDIAN@|$(if $(CONFIG_BIG_ENDIAN),big,little)|" \
 		< $(MESON_DIR)/openwrt-cross.txt.in \
 		> $(1)
